### PR TITLE
[azeventhubs] Export connection string parsing code

### LINKED ISSUE #19853: [azeventhubs] Export a connection string parser
(edited to remove the EventHub prefixes, it'll be in the azeventhubs package already) 

APIView: [apiview](https://apiview.dev/Assemblies/Review/8245669789dd4453bea087a9afaf2705?diffRevisionId=a0e5cec4afa24cdfb0f148a6c74d2572&diffOnly=False&revisionId=8d3115c70fac4463b67fc1469baea81c&doc=False)

This is another part of our track 2 SDKs, which allow users to pass in an Event Hubs connection string so you can get to the individual parts like the namespace or the entity.

Here's a few examples:

* [.NET: EventHubsConnectionStringProperties](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/eventhub/Azure.Messaging.EventHubs/src/EventHubsConnectionStringProperties.cs)
* [Java: EventHubConnectionStringProperties](https://github.dev/Azure/azure-sdk-for-java/blob/f9d9f8f9125934a5644bfb2c985b2029c043087a/sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/models/EventHubConnectionStringProperties.java)
* [Python: EventHubConnectionStringProperties](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/eventhub/azure-eventhub/azure/eventhub/_connection_string_parser.py#L9)
* [JS: EventHubConnectionConfig](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/eventhub/event-hubs/src/eventhubConnectionConfig.ts)

We'll need to discuss the design for Go. Tentatively we can do:

- A "constructor" function to parse a connection string
-A new type to hold onto the attributes

```go
package azeventhubs

type ConnectionStringProperties struct {
  // ex: sb://myeventhub.servicebus.windows.net
  Endpoint string

  // ex: MyEventHub
  EventHubName string

  // ex: myeventhub.servicebus.windows.net
  FullyQualifiedNamespace string

  SharedAccessKey string
  SharedAccessKeyName string
  SharedAccessSignature string
}

func NewConnectionStringProperties(connectionString string) (ConnectionStringProperties, error) {
  // parse the string, as we do internally
  retur ConnectionStringProperties{}
}
```

Should note that we already have this code, this would just be a matter of exporting it in a customer-palatable way.

### PR DESCRIPTION
**Pull request by richardpark-msft**
Exporting a function to allow users to parse Event Hub connection strings.

Fixes #19853, #19840

### CODE REVIEW

File: sdk/messaging/azeventhubs/connection_string_properties.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azeventhubs
+
+import "github.com/Azure/azure-sdk-for-go/sdk/messaging/azeventhubs/internal/exported"
+
+// ConnectionStringProperties are the properties of a connection string
+// as returned by [NewConnectionStringProperties].
```
Comment (jhendrixMSFT) [inline sdk/messaging/azeventhubs/connection_string_properties.go:9]: Also update comment here.

File: sdk/messaging/azeventhubs/internal/exported/connection_string_properties.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"strings"
+)
+
+// ConnectionStringProperties are the properties of a connection string
+// as returned by [NewConnectionStringProperties].
```
Comment (jhendrixMSFT) [inline sdk/messaging/azeventhubs/internal/exported/connection_string_properties.go:14]: Update comment.

File: sdk/messaging/azeventhubs/internal/exported/connection_string_properties.go [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+	// Endpoint is the Endpoint value in the connection string.
+	// Ex: sb://example.servicebus.windows.net
+	Endpoint *string
+
+	// EntityPath is EntityPath value in the connection string.
+	EntityPath *string
+
+	// FullyQualifiedNamespace is the Endpoint value without the protocol scheme.
+	// Ex: example.servicebus.windows.net
+	FullyQualifiedNamespace *string
```
Comment (jhendrixMSFT) [inline sdk/messaging/azeventhubs/internal/exported/connection_string_properties.go:25]: If this, and others, are always required then IMO they shouldn't be pointer-to-type.

File: sdk/messaging/azeventhubs/CHANGELOG.md [hunk 4]
```diff
@@ -5,6 +5,7 @@
 ### Features Added
 
 - Adds ProcessorOptions.Prefetch field, allowing configuration of Prefetch values for PartitionClients created using the Processor. (PR#19786)
+- Added new function to parse connection string into values using `NewConnectionStringProperties` and `ConnectionStringProperties`. (PR#19855)
```
[the remark this answers is not in the document]
Comment (richardpark-msft) [reply sdk/messaging/azeventhubs/CHANGELOG.md]: Done.

File: sdk/messaging/azeventhubs/connection_string_properties.go (hunk 1 above)
Comment (richardpark-msft) [reply sdk/messaging/azeventhubs/connection_string_properties.go]: Done.

File: sdk/messaging/azeventhubs/internal/exported/connection_string_properties.go (hunk 2 above)
Comment (richardpark-msft) [reply sdk/messaging/azeventhubs/internal/exported/connection_string_properties.go]: Done.

File: sdk/messaging/azeventhubs/internal/exported/connection_string_properties.go (hunk 3 above)
Comment (richardpark-msft) [reply sdk/messaging/azeventhubs/internal/exported/connection_string_properties.go]: Done. FullyQualifiedNamespace and Endpoint are the only ones that are guaranteed to be there - the others are mutually exclusive so we'll leave those as pointerse.

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**